### PR TITLE
docs: 📚 improve developer docu

### LINKED ISSUE #356: docs: 📚 improvement of developer docu
# Description

### Framwork wrapper
everything which is stated here should be done for every framework:

- add the explanation for the naming convention for each framework (e.g. in Vue we always have `SynVue` as prefix for the component name and the  rest is added via camelcase: syn-button = SynVueButton, syn-icon-button = SynVueIconButton)
- [x] angular
- [x] vue
- [x] react

- add explicit link to our demo repository with the information, that an example usage in this framework can be seen there
- [x] angular
- [x] vue
- [x] react

- add example for an input with e.g. SynChangeEvent and usage of the `target.value` for typescript
```
onChange={( e: SynChangeEvent) => someMethod((e.target as SynInput).value)}
```
- [x] angular
- [x] vue
- [x] react

- add example with component from the framework wrapper package and type of the components package
```
import {SynButton} from '@synergy-design-system/react';
import {type SynButton as SynButtonType} from '@synergy-design-system/components';

const buttonRef = useRef<SynButtonType>(null)
<SynButton ref={buttonRef}>
My Button
</SynButton>
```
- [x] angular
- [x] vue
- [x] react

### Angular wrapper
- [x] add angular 16 to the compatibility statement: `We are currently supporting Angular version 16 and 17.`
- [x] add information, that typescript version >=5 is needed, otherwise there will be typescript compilation errors

### Icons
- [x] nowhere is explained, that the assets-package needs to be installed and how it is installed. this should be added at the icon story documentation
```
npm i @synergy-design-system/assets
```
- [x] add readme and changelog of assets package to storybook

### Icon-button
- [x] nowhere is explained, that the assets-package needs to be installed and how it is installed. this should be added at the icon story documentation (maybe just a link to the icon docu)

### General
- [x] add figma links to all components / packages

# Rationale

As a user I want a good documentation, so it`s easy for me to get started with synergy

### PR DESCRIPTION
**Pull request by kirchsuSICKAG**
# Pull Request

## 📖 Description

This PR improves the developer documentation like framework wrapper Readme's, adding corresponding Figma Links to all components, improve assets package description, etc.

### 🎫 Issues

Closes #356 and #362 

## 👩‍💻 Reviewer Notes

Have a look at: 
- the framework wrapper Readme's
- how the changelogs are looking now
- the assets Readme
- if the figma links are working correctly (Design tab in storybook documentation)
- the adapted docs token of the default icon-button story in `packages/tokens/src/figma-tokens/_docs.json`

>Note: it was not possible to change the order of the Categories (Attributes, Events, Slots, ...) in the Storybook 'Docs' page table or close them as default with only resonable time spending / effort

## ✅ DoD

- [x] I have tested my changes manually.
- [ ] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have made sure to follow the projects coding and contribution guides.
- [ ] I have made sure that the bundle size has changed appropriately.
- [ ] I have validated that there are no accessibility errors.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

File: packages/assets/README.md [hunk 1]
```diff
@@ -1,4 +1,40 @@
-## Documentation
+# @synergy-design-system/assets
+
+This package provides assets like:
+
+- **logos:** the variants of the SICK brand logo
```
Comment (klaarseSICKAG) [inline packages/assets/README.md:5]: isthe definition valid for the future too? I could imagine adding other logos as well. so maybe we should be more generic like (such as SICK brand logo)

File: packages/react/README.md [hunk 2]
```diff
@@ -46,23 +49,65 @@ createRoot(document.getElementById("root")!).render(
...
+```
+
+```tsx
+// React wrapper example
+<SynButton> My Button </SynButton>
+```
+
+### 5. Usage of attributes
+
+The attribute naming of the components are the same as in the documentation.
```
Comment (klaarseSICKAG) [inline packages/react/README.md:82]:
```suggestion
The attribute namings of the components are the same as in the documentation.
```

File: packages/vue/README.md [hunk 3]
```diff
@@ -54,9 +57,120 @@ You may now use the components by importing them from the `@synergy-design-syste
...
+</script>
+
+<template>
+  <SynVueInput label="Surname" @syn-change="synChange" />
+</template>
+```
+
+### 7. Usage of methods
+
+Components can have methods (like `focus`, `click`, `stepUp`, etc. ), which can trigger an action, if they are called.
```
Comment (klaarseSICKAG) [inline packages/vue/README.md:144]:
```suggestion
Components can have methods (like `focus`, `click`, `stepUp`, etc. ), which trigger an action, if they are called.
```

**Review by klaarseSICKAG — COMMENTED** [review]
Nice work!
Lets see how storybook looks like...

File: packages/angular/README.md [hunk 4]
```diff
@@ -21,15 +21,18 @@ Run the following steps to install the required packages.
...
+# If not already installed, install Angular's peer dependencies
 # Install step for angular@17
 npm install --save @angular/core@17 @angular/forms@17
+
+# Optional: if icons shall be used, install the assets package
+npm install --save @synergy-design-system/assets
 ```
 
-### 2. Add the desired theme to your application
+### 2. Add the wanted theme to your application
```
[the remark this answers is not in the document]
Comment (kirchsuSICKAG) [reply packages/angular/README.md:32]: because in the other two frameworks (vue and react) it is `wanted`. But I can change it for all to `desired`

**Comment by klaarseSICKAG** [comment]
- new changelog formatting looks great 👍
- figma links are just amazing 🤩

**Review by klaarseSICKAG — APPROVED** [review]

**Pull request merged**